pytensor/link/pytorch/linker.py

```diff
@@ -18,14 +24,45 @@ def output_filter(self, var: Variable, out: Any) -> Any:
     def fgraph_convert(self, fgraph, input_storage, storage_map, **kwargs):
         from pytensor.link.pytorch.dispatch import pytorch_funcify
 
+        # We want to have globally unique names
+        # across the entire pytensor graph, not
+        # just the subgraph
+        generator = unique_name_generator(["torch_linker"])
+        built_kwargs = {"unique_name": generator, "linker": self, **kwargs}
         return pytorch_funcify(
-            fgraph, input_storage=input_storage, storage_map=storage_map, **kwargs
+            fgraph, input_storage=input_storage, storage_map=storage_map, **built_kwargs
         )
 
     def jit_compile(self, fn):
         import torch
 
-        return torch.compile(fn)
+        class wrapper:
```

Comment:
add some docstrings explaining why we're doing this and that we can hopefully revert in the future? Link to issue on torch if you opened one already